### PR TITLE
optimize init and fix tracy

### PR DESCRIPTION
**Pull request by Kapu1178**
## Changelog

:cl:

/:cl:

### CODE REVIEW

File: code/game/turfs/open/space/space.dm [hunk 1]
```diff
@@ -58,7 +58,7 @@ GLOBAL_REAL_VAR(space_appearances) = make_space_appearances()
 
 	initialized = TRUE
 
-	if(!loc:area_has_base_lighting) //Only provide your own lighting if the area doesn't for you
+	if(!loc.luminosity) //Only provide your own lighting if the area doesn't for you
```
Comment (LemonInTheDark) [inline code/game/turfs/open/space/space.dm:61]: not a safe assumption areas default to lumin on, and non lit areas can have lumin, see area/New (unless we've drifted)

File: code/modules/lighting/lighting_turf.dm [hunk 2]
```diff
@@ -53,8 +53,7 @@
 	if (!lighting_object)
 		return FALSE
 
-	return !(luminosity || dynamic_lumcount)
-
+	return !(loc:area_has_base_lighting || luminosity || dynamic_lumcount)
```
Comment (LemonInTheDark) [inline code/modules/lighting/lighting_turf.dm:56]: should just add a getter for a turf's effective lumin

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/game/turfs/open/space/space.dm (hunk 1 above)
Comment (Kapu1178) [reply code/game/turfs/open/space/space.dm:61]: When would it ever be valid for an area without base lighting to have luminosity? Seems to me like they should be intrinsically paired during `Initialize()`

File: code/modules/lighting/lighting_turf.dm (hunk 2 above)
Comment (Kapu1178) [reply code/modules/lighting/lighting_turf.dm]: Eh, it's one place.

**Pull request merged**